lib/sorbet-schema/hash_transformer.rb

```diff
@@ -9,7 +9,14 @@ class HashTransformer
   sig { params(hash: T::Hash[T.untyped, T.untyped]).returns(T::Hash[Symbol, T.untyped]) }
   def deep_symbolize_keys(hash)
     hash.each_with_object({}) do |(key, value), result|
-      result[key.to_sym] = value.is_a?(Hash) ? deep_symbolize_keys(value) : value
+      result[key.to_sym] = unpack_value(value)
+    end
+  end
+
+  sig { params(hash: T::Hash[T.untyped, T.untyped]).returns(T::Hash[Symbol, T.untyped]) }
+  def deep_symbolize_keys_serialize_values(hash)
```

Comment:
suggestion(blocking): I think this needs to remain two separate operations, one that operates on keys, and one that operates on values. We can call both from outside, but I'd prefer the call site to look like:

`serialize_values(deep_symbolize_keys(hsh))`

Even though that looks weird, I like keeping the concerns separated.

Reply:
We decided instead of separation into different methods, we'd just have one method that controls whether the values are serialized.